src/assemble_workflow/bundle.py

```diff
@@ -41,10 +41,11 @@ def __init__(self, build_manifest: BuildManifest, artifacts_dir: str, bundle_rec
         :param bundle_recorder: The bundle recorder that will capture and build a BundleManifest
         """
         self.build = build_manifest.build
-        self.plugins = self.__get_plugins(build_manifest.components)
+        self.components = build_manifest.components
         self.artifacts_dir = artifacts_dir
         self.bundle_recorder = bundle_recorder
         self.tmp_dir = TemporaryDirectory(keep=keep)
+        self.min_bundle = self.__get_min_bundle(build_manifest.components)
```

Comment:
They look similar by name but are actually two different objects.

`__get_min_bundle` is the component corresponding to Core inside the build manifest. To visualize this, it is the one in the red box.
<img width="993" alt="Screen Shot 2022-03-07 at 12 20 47 AM" src="https://user-images.githubusercontent.com/60111637/156994271-4b010100-3b3a-4740-87cc-d468e141c3b3.png">

This is the reason why this logic works by skipping the core component during installation.

```
           if self.min_bundle == c:
                pass
```

`__get_min_dist` is a `Dist` object https://github.com/opensearch-project/opensearch-build/blob/main/src/assemble_workflow/dist.py Depending on what is specified in the manifest `distribution` field, it could be a `DistTar` https://github.com/opensearch-project/opensearch-build/blob/main/src/assemble_workflow/dist.py#L73

or a `DistZip` https://github.com/opensearch-project/opensearch-build/blob/main/src/assemble_workflow/dist.py#L83

This is the reference to the `distribution` field.
<img width="771" alt="Screen Shot 2022-03-07 at 12 28 04 AM" src="https://user-images.githubusercontent.com/60111637/156995075-73827ef9-7f0d-4d7e-9d6c-ce62b1826b17.png">

Reply:
Got it. Thanks for explaining